src/mbgl/util/thread_context.cpp

```diff
@@ -1,30 +1,95 @@
 #include <mbgl/util/thread_context.hpp>
 
+#include <mbgl/util/thread_local.hpp>
+
+#include <cassert>
+
+namespace {
+
+using namespace mbgl::util;
+static tls<ThreadContext>& current = *new tls<ThreadContext>;
```

Comment:
Yes and I did the same with `RunLoop`. I was hitting sometimes a bug that would sometimes the `static tls<RunLoop> current` before a `RunLoop` object on the heap. The object on the heap sets the `::current` to `nullptr` and crashes.

That reminded me that a practice we had at the Chromium browser to "leak" static variables to prevent this sort of issues: https://code.google.com/p/chromium/codesearch#chromium/src/base/macros.h&l=153

It is not exactly a leak because it is still referenced (so valgrind won't complain) and it would only be destructed at exit anyway.